src/pudl/transform/ferc1.py

```diff
@@ -2020,43 +2807,53 @@ class PlantInServiceFerc1TableTransformer(Ferc1AbstractTableTransformer):
     def process_xbrl_metadata(self, xbrl_metadata_json) -> pd.DataFrame:
         """Transform the metadata to reflect the transformed data.
 
-        The XBRL Taxonomy metadata as extracted pertains to the XBRL data as extracted.
-        When we re-shape the data, we also need to adjust the metadata to be usable
-        alongside the reshaped data. For the plant in service table, this means
-        selecting metadata fields that pertain to the "stem" column name (not
-        differentiating between starting/ending balance, retirements, additions, etc.)
-
         We fill in some gaps in the metadata, e.g. for FERC accounts that have been
         split across multiple rows, or combined without being calculated. We also need
         to rename the XBRL metadata categories to conform to the same naming convention
         that we are using in the data itself (since FERC doesn't quite follow their own
         naming conventions...). We use the same rename dictionary, but as an argument to
         :meth:`pd.Series.replace` instead of :meth:`pd.DataFrame.rename`.
+
+        We also deduplicate the metadata on the basis of
         """
-        pis_meta = (
-            super()
-            .process_xbrl_metadata(xbrl_metadata_json)
-            .assign(
-                xbrl_factoid=lambda x: x.xbrl_factoid.replace(
-                    self.params.rename_columns_ferc1.instant_xbrl.columns
-                )
-            )
-        )
+        tbl_meta = super().process_xbrl_metadata(xbrl_metadata_json)
 
         # Set pseudo-account numbers for rows that split or combine FERC accounts, but
         # which are not calculated values.
-        pis_meta.loc[
-            pis_meta.xbrl_factoid == "electric_plant_purchased", "ferc_account"
+        tbl_meta.loc[
+            tbl_meta.xbrl_factoid == "electric_plant_purchased", "ferc_account"
         ] = "102_purchased"
-        pis_meta.loc[
-            pis_meta.xbrl_factoid == "electric_plant_sold", "ferc_account"
+        tbl_meta.loc[
+            tbl_meta.xbrl_factoid == "electric_plant_sold", "ferc_account"
         ] = "102_sold"
-        pis_meta.loc[
-            pis_meta.xbrl_factoid
+        tbl_meta.loc[
+            tbl_meta.xbrl_factoid
             == "electric_plant_in_service_and_completed_construction_not_classified_electric",
             "ferc_account",
         ] = "101_and_106"
-        return pis_meta
+
+        # remove duplication of xbrl_factoid
+        same_calcs_mask = tbl_meta.duplicated(
+            subset=["xbrl_factoid", "calculations"], keep=False
+        )
+        # if they key values are the same, select the records with values in ferc_account
+        same_calcs_deduped = (
+            tbl_meta[same_calcs_mask]
+            .sort_values(["ferc_account"])
+            .drop_duplicates(subset=["xbrl_factoid"], keep="first")
```

Comment:
it could but it doesn't need to

Reply:
But it's going to generate a potentially different set of duplicates if it's using different parameters isn't it?

Reply:
i understand what you are saying and i could add it in but i think it is less intuitive and it is not needed.

the first line here is grabbing the records with the same calcs & same factoid names. but the the column that ultimately we really want to be de-duping based on is the factoid. so that is what we are selecting on.